spring-integration-core/src/main/java/org/springframework/integration/transaction/DefaultTransactionSynchronizationFactory.java

```diff
@@ -46,13 +49,16 @@ public DefaultTransactionSynchronizationFactory(TransactionSynchronizationProces
 	@Override
 	public TransactionSynchronization create(Object key) {
 		Assert.notNull(key, "'key' must not be null");
-		DefaultTransactionalResourceSynchronization synchronization = new DefaultTransactionalResourceSynchronization(key);
-		TransactionSynchronizationManager.bindResource(key, synchronization.getResourceHolder());
-		return synchronization;
+		if (!TransactionSynchronizationManager.hasResource(key)) {
```

Comment:
I can live with it if we deprecate and rename the method to something like `createAndBindSynchronizationIfNeeded`.